x-pack/plugins/search_indices/public/hooks/api/use_delete_document.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AcknowledgedResponseBase } from '@elastic/elasticsearch/lib/api/types';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { SearchHit } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { QueryKeys } from '../../constants';
+import { useKibana } from '../use_kibana';
+import { INDEX_SEARCH_POLLING, IndexDocuments } from './use_document_search';
+
+export const useDeleteDocument = (indexName: string) => {
+  const { http } = useKibana().services;
+  const queryClient = useQueryClient();
+
+  const result = useMutation(
+    async ({ id }: { id: string }) => {
+      const response = await http.delete<AcknowledgedResponseBase>(
+        `/internal/search_indices/${indexName}/documents/${id}`
+      );
+      return response.acknowledged;
+    },
+    {
+      onMutate: async ({ id }: { id: string }) => {
+        await queryClient.cancelQueries([QueryKeys.SearchDocuments, indexName]);
+
+        const previousData = queryClient.getQueryData<IndexDocuments>([
+          QueryKeys.SearchDocuments,
+          indexName,
+        ]);
+
+        queryClient.setQueryData(
+          [QueryKeys.SearchDocuments, indexName],
+          (snapshot: IndexDocuments | undefined) => {
+            const oldData = snapshot ?? { results: { data: [] } };
+            return {
+              ...oldData,
+              results: {
+                ...oldData.results,
+                data: oldData.results.data.filter((doc: SearchHit) => doc._id !== id),
+              },
+            } as IndexDocuments;
+          }
+        );
+
+        return { previousData };
+      },
+
+      onSuccess: () => {
+        setTimeout(() => {
```

Comment:
this code is particularly smelly - i had to set a timeout to delay the refetch for 5 seconds until hopefully ES3 prop. This delay wont be helpful for stack either. Without the timeout, the result will pop back into view and then be removed once our polling retriggers a fetch. 

Open to better suggestions here

Reply:
Is it happening because of the refresh delay on ES3? If that is the case it is fine for now until we get a proper fix. Otherwise I would look for an option to retrigger a fetch but looks like it is what you are doing. 

Ideally I would show an indicator until we are sure the next poll is happened, and invalidate query for fetching the search documents.